### ISSUE TITLE
how to use certain classes of Coco Dataset

### ISSUE THREAD

**Issue by moahaimen** [ultralytics/ultralytics#3963]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

hi @glenn-jocher 
I need to train my Yolov8 only on 5 classes of the COCO dataset i don't want to download all the dataset, i use google colab, and i used the coco.yaml

/content/yolov8/ultralytics/cfg/datasets/coco.yaml

# Ultralytics YOLO 🚀, AGPL-3.0 license
# COCO 2017 dataset http://cocodataset.org by Microsoft
# Example usage: yolo train data=coco.yaml
# parent
# ├── ultralytics
# └── datasets
#     └── coco  ← downloads here (20.1 GB)

# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: ../datasets/coco  # dataset root dir
train: train2017.txt  # train images (relative to 'path') 118287 images
val: val2017.txt  # val images (relative to 'path') 5000 images
test: test-dev2017.txt  # 20288 of 40670 images, submit to https://competitions.codalab.org/competitions/20794

# Classes
names:
  
  0: bird
  1: cat
  2: cup
  3: knife
  4: apple
  5: mouse
  
# Download script/URL (optional)
download: |
  from ultralytics.utils.downloads import download
  from pathlib import Path

  # Download labels
  segments = True  # segment or box labels
  dir = Path(yaml['path'])  # dataset root dir
  url = 'https://github.com/ultralytics/yolov5/releases/download/v1.0/'
  urls = [url + ('coco2017labels-segments.zip' if segments else 'coco2017labels.zip')]  # labels
  download(urls, dir=dir.parent)
  # Download data
  urls = ['http://images.cocodataset.org/zips/train2017.zip',  # 19G, 118k images
          'http://images.cocodataset.org/zips/val2017.zip',  # 1G, 5k images
          'http://images.cocodataset.org/zips/test2017.zip']  # 7G, 41k images (optional)
  download(urls, dir=dir / 'images', threads=3)

i specified only the classes i need but when i execute the training, 
from ultralytics.yolo.engine.model import YOLO
# # Load a model
model = YOLO("yolov8s.pt")  # load a pretrained model (recommended for training)
#
# # Use the model
model.train(data="coco.yaml", epochs=150)  # 

but also the whole dataset downloaded, and training didnt complete

**Comment by glenn-jocher** [comment]
@moahaimen hello,

Thanks for your specific question on training the YOLOv8 model with a subset of classes from the COCO dataset.

In the current setup of the Ultralytics YOLOv8 repository, the model training process downloads the entire COCO dataset even if only a few classes are being used. This behavior is intended as it simplifies the setup process for most users.

However, if you're only interested in a few classes and you want to avoid downloading the whole dataset, you would have to create a customized dataset containing only your classes of interest. This would likely involve downloading the dataset to a local environment, extracting the images and annotations for the specific classes you're interested in, and then employing this customized dataset for your training purposes.

Regarding the incomplete training issue, it could be due to various reasons, such as insufficient system resources, particularly memory. But it's challenging to diagnose without specific error messages or logs.

Remember that training a deep learning model, especially a complex one like YOLOv8, can be resource-intensive. Thus, ensure that your Google Colab environment has enough resources (like GPU memory), and that your runtime session stays connected throughout the entire training process.

I hope this helps! Let me know if you have additional questions.

**Comment by moahaimen** [comment]
@glenn-jocher  thank you for responding, dear sir due to the hardware needed to work on the coco dataset, i just need your help in knowing the mAP0.5 and the mAP0.5:0.95 and Precision and Recall  for the YOLOV5 and Yolov8 ,for the 300 epochs   
i wish your help

**Comment by glenn-jocher** [comment]
@moahaimen hello,

Thank you for your inquiry about the metrics for YOLOv8. 

The values for mAP 0.5 and mAP 0.5:0.95, as well as the precision and recall, can vary from training to training due to factors such as parameter initialization, fluctuating training circumstances, and other stochastic elements inherent in deep learning algorithms. That said, the expected values should generally fall within a predictable range.

However, in the scope of this YOLOv8 repository, we only provide information and assistance regarding the YOLOv8 model, so we are not able to provide any comparative statistics for YOLOv5 or any other models outside the YOLOv8 scope.

For detailed, accurate and up-to-date performance metrics specifically for YOLOv8 over 300 epochs, may I suggest you refer to the official YOLOv8 documentation and possibly re-run the benchmark tests or model training in your own environment to obtain the most relevant and precise results?

Remember to always have patience in training such complex deep learning models, as results can sometimes vary greatly and require intensive computational resources, even at a standard level such as 300 epochs.

I hope this information was helpful and if you have further queries or issues, feel free to ask.
